src/derivers/bip32.ts

```diff
@@ -1,56 +1,83 @@
 import { keccak_256 as keccak256 } from '@noble/hashes/sha3';
 import { hmac } from '@noble/hashes/hmac';
 import { sha512 } from '@noble/hashes/sha512';
-import { BUFFER_KEY_LENGTH, BIP_32_HARDENED_OFFSET } from '../constants';
+import { BIP_32_HARDENED_OFFSET, BUFFER_KEY_LENGTH } from '../constants';
 import { bytesToNumber, hexStringToBuffer, isValidBufferKey } from '../utils';
 import { Curve, mod, secp256k1 } from '../curves';
 
+export function privateKeyToEthAddress(key: Buffer) {
+  if (!Buffer.isBuffer(key) || !isValidBufferKey(key, 32)) {
+    throw new Error('Invalid key: The key must be a 32-byte, non-zero Buffer.');
+  }
+
+  const publicKey = secp256k1.getPublicKey(key, false);
+  return publicKeyToEthAddress(publicKey);
+}
+
 /**
- * Converts a BIP-32 private key to an Ethereum address.
+ * Converts a BIP-32 public key to an Ethereum address.
  *
  * **WARNING:** Only validates that the key is non-zero and of the correct
  * length. It is the consumer's responsibility to ensure that the specified
  * key is a valid BIP-44 Ethereum `address_index` key.
  *
- * @param key - The `address_index` key buffer to convert to an Ethereum
+ * @param key - The `address_index` public key buffer to convert to an Ethereum
  * address.
  * @returns The Ethereum address corresponding to the given key.
  */
-export function privateKeyToEthAddress(key: Buffer) {
-  if (!Buffer.isBuffer(key) || !isValidBufferKey(key)) {
-    throw new Error('Invalid key: The key must be a 64-byte, non-zero Buffer.');
+export function publicKeyToEthAddress(key: Buffer) {
+  if (!Buffer.isBuffer(key) || !isValidBufferKey(key, 65)) {
+    throw new Error('Invalid key: The key must be a 65-byte, non-zero Buffer.');
   }
 
-  const privateKey = key.slice(0, 32);
-  const publicKey = secp256k1.getPublicKey(privateKey, false).slice(1);
-  return Buffer.from(keccak256(Buffer.from(publicKey)).slice(-20));
+  return Buffer.from(keccak256(key.slice(1)).slice(-20));
 }
 
 /**
  * @param pathPart
  * @param parentKey
+ * @param parentPublicKey
+ * @param chainCode
  * @param curve
  */
 export async function deriveChildKey(
   pathPart: string,
-  parentKey: Buffer,
+  parentKey: Buffer | undefined,
```

Comment:
`parentKey?: Buffer`?

Reply:
Can't do that here since it's not the last parameter and some parameters after it are required.